services/admin/src/store.ts

```diff
@@ -360,23 +405,23 @@ export class Store {
         }
       }
 
-      this.client.run(
-        analyzeOnly ? 'rollback transaction' : 'commit transaction'
-      );
-      inTransaction = false;
-
       return ok({
         id,
         wasExistingFile: !!existing,
         newlyAdded,
         alreadyPresent,
       });
-    } catch (error) {
-      if (inTransaction) {
-        this.client.run('rollback transaction');
+    };
+
+    function shouldCommit(result: AddCastVoteResult) {
+      if (analyzeOnly) {
+        return false;
       }
-      throw error;
+
+      return result.isOk();
```

Comment:
Maybe this reads easier?
```suggestion
      return !analyzeOnly && result.isOk();
```

Or even inline it?
```ts
return await this.client.transaction(transactionFn, (result) => !analyzeOnly && result.isOk());
```

Reply:
ha! funny thing is, i had it exactly that way before and decided to expand it for readability over conciseness - i can go either way

Reply:
I don't feel strongly. I do like that the way you had it keeps the name which serves as a hint as to its function.